openshift-knative-operator/pkg/monitoring/rbac_proxy_injection.go

```diff
@@ -32,6 +32,13 @@ func InjectRbacProxyContainerToDeployments(deployments sets.String) mf.Transform
 			firstContainer := &dep.Spec.Template.Spec.Containers[0]
 			// Make sure we export metrics only locally
 			firstContainer.Env = append(firstContainer.Env, corev1.EnvVar{Name: "METRICS_PROMETHEUS_HOST", Value: "127.0.0.1"})
+			if u.GetName() == "kafka-ch-controller" {
```

Comment:
why the special case for the controller?

Reply:
@skonto should we add a comment? on the why?

Reply:
Because only that deployment needs to know the value of `EnableMonitoringEnvVar` because it replicated the logic of S-O in order to manage the dispatcher's monitoring setup. I can add a comment.

Reply:
done